### PR TITLE
Add CAKE build script

### PR DESCRIPTION
**Pull request by cra0zy**
Adds the build script as per #6773.

Just run either `build.sh` or `build.ps1` depending on the platform.

Stuff will slowly be added to the build script as they are moved from Protobuild.

### CODE REVIEW

**Comment by Jjagg** [comment]
Could you use DotNetCorePack() instead of MSBuild() for packing? It's more consistent and will remove the MSBuild dependency.
https://cakebuild.net/api/Cake.Common.Tools.DotNetCore/DotNetCoreAliases/0F7A518E

**Comment by jnoyola** [comment]
> Could you use DotNetCorePack() instead of MSBuild() for packing?

Just reading through. He mentioned the following in https://github.com/MonoGame/MonoGame/pull/6772#issuecomment-505391103, so I'm assuming it needs to be MSBuild().
> dotnet msbuild does not support standard .net framework compilation on non Windows platforms

**Comment by Jjagg** [comment]
Ah, thanks. I missed that comment.

**Comment by jnoyola** [comment]
Cake looks pretty powerful overall, but it looks like right now it'll be used here for restoring, building, and packing. Is this difficult to orchestrate cross-platform with just dotnet and msbuild, or do we expect to see the full benefit of Cake as the complexity increases in the future?

**Comment by cra0zy** [comment]
Unit testing, template packing, installer building...

**Comment by Jjagg** [comment]
The MSBuild call fails on Windows:
```
\MonoGame\MonoGame.Framework\MonoGame.Framework.DesktopGL.csproj(1,1): error MSB4041: 
  The default XML namespace of the project must be the MSBuild XML namespace.
  If the project is authored in the MSBuild 2003 format, please add 
  xmlns="http://schemas.microsoft.com/developer/msbuild/2003"  to the <Project> element.
  If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.
An error occurred when executing task 'BuildDesktopGL'.
```

File: build.cake [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    dnpacksettings.Configuration = configuration;
+});
+
+Task("BuildDesktopGL")
+    .IsDependentOn("Prep")
+    .Does(() =>
+{
+    DotNetCoreRestore("MonoGame.Framework.DesktopGL.sln");
+
+    if (IsRunningOnWindows())
```
Comment (Jjagg) [inline build.cake]: Maybe a short comment to explain why this is necessary?
Also would be nice to make a function for this to make it less verbose:
```
void PackProject(string project)
{
    DotNetCoreRestore(project);

    if (IsRunningOnWindows())
        DotNetCorePack(project, dnpacksettings);
    else
        MSBuild(project, mspacksettings);
}
```

**Comment by Jjagg** [comment]
Looks good, merging!

**Pull request merged**